dev_docs/tutorials/saved_objects.mdx

```diff
@@ -241,10 +243,11 @@ export const dashboardVisualization: SavedObjectsType = {
  in which this migration was released. So if you are creating a migration which will
  be part of the v7.10.0 release, but will also be backported and released as v7.9.3, the migration version should be: 7.9.3.
 
- Migrations should be written defensively, an exception in a migration function will prevent a Kibana upgrade from succeeding and will cause downtime for our users. 
- Having said that, if a 
- document is encountered that is not in the expected shape, migrations are encouraged to throw an exception to abort the upgrade. In most scenarios, it is better to
-  fail an upgrade than to silently ignore a corrupt document which can cause unexpected behaviour at some future point in time.
+Migrations should be written defensively, an exception in a migration function will prevent a Kibana upgrade from succeeding and will cause downtime for our users. 
+Having said that, if a document is encountered that is not in the expected shape, migrations are encouraged to throw an exception to abort the upgrade. In most scenarios, it is better to
+fail an upgrade than to silently ignore a corrupt document which can cause unexpected behaviour at some future point in time. When such a scenario is encountered,
+the error should be verbose and informative so that the corrupt document can be corrected, if possible.
+
+### Testing Migrations
 
-It is critical that you have extensive tests to ensure that migrations behave as expected with all possible input documents. Given how simple it is to test all the branch 
-conditions in a migration function and the high impact of a bug in this code, there’s really no reason not to aim for 100% test code coverage.
+See the <DocLink id="kibDevTutorialTestingPlugins" section="saved-object-migrations" text="Saved Object migrations section in the plugin testing guide"/>.
```

Comment:
nit: just a small paragraph so that this link doesn't seem like optional reading...

```suggestion
Bugs in a migration function cause downtime for our users and therefore have a very high impact. Follow the <DocLink id="kibDevTutorialTestingPlugins" section="saved-object-migrations" text="Saved Object migrations section in the plugin testing guide"/>.
```